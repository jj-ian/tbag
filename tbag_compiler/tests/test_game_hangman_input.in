M
D
T
I
E
H
B
C
R
F
G
K
