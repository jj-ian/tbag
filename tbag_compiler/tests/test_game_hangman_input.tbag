boolean started = false;
string[3] answer;
int[26] guessed;
int numberOfGuesses = 0;

NOT started {
	
	answer[0] = "C";
	answer[1] = "A";
	answer[2] = "T";

	started = true;
}

numberOfGuesses > arrLen(answer) {
	strPrintLine("ur out of guesses");
	endgame;
}

true {


	int i = 0;
	while (i < arrLen(guessed)) {
		print(guessed[i]);
		print(" ");
		i = i + 1;
	}

	print("Guess a letter.");
	getInputFromOptions("A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P", "Q", "R", "S", "T", "U", "V", "W", "X", "Y", "Z");


		if (isLetterInAnswer(input)) {
			strPrintLine("letter is in answer");
		} else {
			strPrintLine("u fucked up lol");
		}
 
	guessed[intFromLetter(input)] = 1;
	numberOfGuesses = numberOfGuesses + 1;


}



#import stdlib

func boolean hasLetterBeenGuessedBefore(string letter) {
	if (guessed[intFromLetter(letter)] == 1) {
		return true;
	} else {
		return false;
	}

}

func boolean isLetterInAnswer(string letter) {
	int i = 0;
	boolean found = false;
	while (i < arrLen(answer) AND found == false) {
		if (answer[i] ~~ letter) {
			found = true;
		} else {}
		i = i + 1;
	}
	return found;
}
