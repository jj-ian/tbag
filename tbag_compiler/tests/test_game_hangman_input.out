Guess a letter. Choose from one of the following options:
A    B    C    D    E    F    G    H    I    J    K    L    M    N    O    P    Q    R    S    T    U    V    W    X    Y    Z    
Input: M





 _________     
|         |    
|         O    
|           
|              
|              
|              



_ _ _ _ _ _ _ _ _ _ 

Already Guessed : M 


Guess a letter. Choose from one of the following options:
A    B    C    D    E    F    G    H    I    J    K    L    M    N    O    P    Q    R    S    T    U    V    W    X    Y    Z    
Input: D





 _________     
|         |    
|         O    
|        /   
|              
|              
|              



_ _ _ _ _ _ _ _ _ _ 

Already Guessed : D M 


Guess a letter. Choose from one of the following options:
A    B    C    D    E    F    G    H    I    J    K    L    M    N    O    P    Q    R    S    T    U    V    W    X    Y    Z    
Input: T





 _________     
|         |    
|         O    
|        /   
|              
|              
|              



_ _ T _ _ _ _ _ _ _ 

Already Guessed : D M T 


Guess a letter. Choose from one of the following options:
A    B    C    D    E    F    G    H    I    J    K    L    M    N    O    P    Q    R    S    T    U    V    W    X    Y    Z    
Input: I





 _________     
|         |    
|         O    
|        /   
|              
|              
|              



_ I T _ _ _ I _ _ _ 

Already Guessed : D I M T 


Guess a letter. Choose from one of the following options:
A    B    C    D    E    F    G    H    I    J    K    L    M    N    O    P    Q    R    S    T    U    V    W    X    Y    Z    
Input: E





 _________     
|         |    
|         O    
|        /   
|              
|              
|              



_ I T _ _ _ I _ E _ 

Already Guessed : D E I M T 


Guess a letter. Choose from one of the following options:
A    B    C    D    E    F    G    H    I    J    K    L    M    N    O    P    Q    R    S    T    U    V    W    X    Y    Z    
Input: H





 _________     
|         |    
|         O    
|        /   
|              
|              
|              



H I T _ H H I _ E _ 

Already Guessed : D E H I M T 


Guess a letter. Choose from one of the following options:
A    B    C    D    E    F    G    H    I    J    K    L    M    N    O    P    Q    R    S    T    U    V    W    X    Y    Z    
Input: B





 _________     
|         |    
|         O    
|        /|   
|              
|              
|              



H I T _ H H I _ E _ 

Already Guessed : B D E H I M T 


Guess a letter. Choose from one of the following options:
A    B    C    D    E    F    G    H    I    J    K    L    M    N    O    P    Q    R    S    T    U    V    W    X    Y    Z    
Input: C





 _________     
|         |    
|         O    
|        /|   
|              
|              
|              



H I T C H H I _ E _ 

Already Guessed : B C D E H I M T 


Guess a letter. Choose from one of the following options:
A    B    C    D    E    F    G    H    I    J    K    L    M    N    O    P    Q    R    S    T    U    V    W    X    Y    Z    
Input: R





 _________     
|         |    
|         O    
|        /|   
|              
|              
|              



H I T C H H I _ E R 

Already Guessed : B C D E H I M R T 


Guess a letter. Choose from one of the following options:
A    B    C    D    E    F    G    H    I    J    K    L    M    N    O    P    Q    R    S    T    U    V    W    X    Y    Z    
Input: F





 _________     
|         |    
|         O    
|        /|\  
|              
|              
|              



H I T C H H I _ E R 

Already Guessed : B C D E F H I M R T 


Guess a letter. Choose from one of the following options:
A    B    C    D    E    F    G    H    I    J    K    L    M    N    O    P    Q    R    S    T    U    V    W    X    Y    Z    
Input: G





 _________     
|         |    
|         O    
|        /|\  
|        /     
|              
|              



H I T C H H I _ E R 

Already Guessed : B C D E F G H I M R T 


Guess a letter. Choose from one of the following options:
A    B    C    D    E    F    G    H    I    J    K    L    M    N    O    P    Q    R    S    T    U    V    W    X    Y    Z    
Input: K





 _________     
|         |    
|         O    
|        /|\  
|        /     
|              
|              



H I T C H H I K E R 

Already Guessed : B C D E F G H I K M R T 


WOO!!!! YOU WON!!!!!!!!
